### PR TITLE
RS/YJ/RULE19-20

### PR DESCRIPTION
**Pull request by yunjoonjung-PNNL**
This rule hasn't been tested. I'll test it out when the TCD is available.

### CODE REVIEW

**Comment by yunjoonjung-PNNL** [comment]
I didn't import the three functions used in this code on purpose to prevent the CI error.

File: rct229/rulesets/ashrae9012019/section19/section19rule20.py [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+            rmr_context="ruleset_model_instances/0",
+            list_path="$.buildings[*].building_segments[*].heating_ventilating_air_conditioning_systems[*]",
+        )
+
+    def create_data(self, context, data):
+        rmi_b = context.baseline
+        rmi_p = context.proposed
+
+        return {
+            "zone_supply_return_exhaust_relief_terminal_fan_power_dict": get_zone_supply_return_exhaust_relief_terminal_fan_power_dict(
```
[the remark this answers is not in the document]
Comment (yunjoonjung-PNNL) [reply rct229/rulesets/ashrae9012019/section19/section19rule20.py:30]: Good point! Addressed.

**Comment by JacksonJ-KC** [comment]
> This rule hasn't been tested. I'll test it out when the TCD is available.

New PR with rule 19-20 test RMD. https://github.com/pnnl/ruleset-checking-tool/pull/1072

File: rct229/rulesets/ashrae9012019/section19/section19rule20.py [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+
+            proposed_total_supply_fan_power = ZERO.POWER
+            proposed_total_return_fan_power = ZERO.POWER
+            proposed_total_exhaust_fan_power = ZERO.POWER
+            proposed_total_relief_fan_power = ZERO.POWER
+            for zone_id_b in list_zones_served_b:
+                proposed_total_supply_fan_power += (
+                    zone_supply_return_exhaust_relief_terminal_fan_power_dict_p[
+                        zone_id_b
+                    ]["zone_supply_fan_power"]
```
Comment (weilixu) [inline rct229/rulesets/ashrae9012019/section19/section19rule20.py:93]: The function return value is supply_fan_power

zone_supply_return_exhaust_relief_terminal_fan_power_dict[zone["id"]] = {
            "supply_fans_power": zone_total_supply_fan_power,
            "return_fans_power": zone_total_return_fan_power,
            "exhaust_fans_power": zone_total_exhaust_fan_power,
            "relief_fans_power": zone_total_relief_fan_power,
            "terminal_fans_power": zone_total_terminal_fan_power,
        }

File: rct229/rulesets/ashrae9012019/section19/section19rule20.py [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+            expected_baseline_fan_power_supply = calc_vals[
+                "expected_baseline_fan_power_supply"
+            ]
+            expected_baseline_fan_power_return = calc_vals[
+                "expected_baseline_fan_power_return"
+            ]
+            expected_baseline_fan_power_relief = calc_vals[
+                "expected_baseline_fan_power_relief"
+            ]
+            return (
```
Comment (weilixu) [inline rct229/rulesets/ashrae9012019/section19/section19rule20.py:164]: Missing exhaust fan power when taking out the values form calc_vals

File: rct229/rulesets/ashrae9012019/section19/section19rule20.py [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+                "expected_baseline_fan_power_return"
+            ]
+            expected_baseline_fan_power_exhaust = calc_vals[
+                "expected_baseline_fan_power_exhaust"
+            ]
+            expected_baseline_fan_power_relief = calc_vals[
+                "expected_baseline_fan_power_relief"
+            ]
+
+            return f"The calculated system fan power does appear to be distributed to the supply, return, exhaust, and relief fans in the same proportion as the proposed design for {hvac_id_b}.The expected modeled baseline supply, return, exhaust, and relief kW is {expected_baseline_fan_power_supply}, {expected_baseline_fan_power_return}, {expected_baseline_fan_power_exhaust}, {expected_baseline_fan_power_relief} respectively."
```
Comment (weilixu) [inline rct229/rulesets/ashrae9012019/section19/section19rule20.py:199]: Shouldnt this be "The calculated system fan power does **not** appear to be distributed..."?

**Review by weilixu — APPROVED** [review]
Approved with some comments must address

File: rct229/rulesets/ashrae9012019/section19/section19rule20.py (hunk 3 above)
Comment (yunjoonjung-PNNL) [reply rct229/rulesets/ashrae9012019/section19/section19rule20.py:164]: Good catch! Addressed.

File: rct229/rulesets/ashrae9012019/section19/section19rule20.py (hunk 2 above)
Comment (yunjoonjung-PNNL) [reply rct229/rulesets/ashrae9012019/section19/section19rule20.py:93]: Addressed.

File: rct229/rulesets/ashrae9012019/section19/section19rule20.py (hunk 4 above)
Comment (yunjoonjung-PNNL) [reply rct229/rulesets/ashrae9012019/section19/section19rule20.py:199]: Good catch! Addressed.

**Comment by yunjoonjung-PNNL** [comment]
> Approved once the CI error is fixed.

Thank you Weili. I got this corrected!

**Review by charliepnnl — APPROVED** [review]
I just had the one very picky comment regarding the use of "percent" in the RDS.

File: rct229/rulesets/ashrae9012019/section19/section19rule20.py [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+                    ]["relief_fans_power"]
+                )
+
+            total_modeled_fan_power_p = (
+                proposed_total_supply_fan_power
+                + proposed_total_return_fan_power
+                + proposed_total_exhaust_fan_power
+                + proposed_total_relief_fan_power
+            )
+            percent_of_total_supply_p = (
```
[the remark this answers is not in the document]
Comment (yunjoonjung-PNNL) [reply rct229/rulesets/ashrae9012019/section19/section19rule20.py:126]: Good point! Addressed.

**Pull request merged**